### PR TITLE
Esf review fixes

### LINKED ISSUE #15: Review of Elastic Serverless Forwarder Python code for best practices

### PR DESCRIPTION
**Pull request by aspacca**
fixes #15

### CODE REVIEW

**Comment by aspacca** [comment]
missing README.md/CONTRIBUTING.md

File: shippers/factory.py [hunk 1]
```diff
@@ -20,16 +20,16 @@ class ShipperFactory:
     @staticmethod
     def create_from_output(output_type: str, output: Output) -> CommonShipper:
         if output_type == "elasticsearch":
-            elasticsearch_output = ElasticSearchOutput(output.type, output.kwargs)
+            assert isinstance(output, ElasticSearchOutput)
```
Comment (masci) [inline shippers/factory.py:23]: we should avoid `assert`ing at runtime, let's raise an exception instead

File: share/config.py [hunk 2]
```diff
@@ -54,27 +53,13 @@ def __init__(self, output_type: str, kwargs: dict[str, Any]):
         self._dataset: str = ""
         self._namespace: str = ""
```
Comment (masci) [inline share/config.py:54]: let's remove the assignments above, the setters are invoked below and will take care of initializing the underlying fields

**Review by masci — CHANGES_REQUESTED** [review]
Left a couple of comments, overall LGTM thanks for this PR!

File: shippers/factory.py (hunk 1 above)
Comment (aspacca) [reply shippers/factory.py:23]: the assertion is for type hinting so I can access "output.elasticserach_output_params" later on

do you suggest to add an if statement on `isinstance` check and raise an exception?
fine for me as long as I can achieve what I wrote above

**Comment by aspacca** [comment]
@masci fixed

**Review by masci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**